document/json.go

```diff
@@ -72,6 +72,19 @@ func (v Value) String() string {
 	return string(d)
 }
 
+// NewFromDecoder creates a document from a json object json.Decoder
+func NewFromDecoder(dec *json.Decoder) (Document, error) {
```

Comment:
This function is not necessary, see comment above.
Basically, [FieldBuffer ](https://pkg.go.dev/github.com/genjidb/genji@v0.7.1/document#FieldBuffer) implements the right interfaces ([json.Marshaler](https://golang.org/pkg/encoding/json/#Marshaler) and [json.Unmarshaler](https://golang.org/pkg/encoding/json/#Unmarshaler)) which makes it compatible with all the json related functions.